### ISSUE TITLE
Plugins reusePaths and removeXMLNS conflict as of version 2.8.0

### ISSUE THREAD

**Issue by remcohaszing** [svg/svgo#1617]
**Describe the bug**

As of version 2.8.0, combining both `removeXMLNS` and `reusePaths` yields an error.

**To Reproduce**

Given the following script:

```js
const { optimize } = require('svgo');

const options = {
  js2svg: {
    indent: 2,
    pretty: true,
  },
  multipass: true,
  plugins: [
    { name: 'reusePaths', active: true },
    { name: 'removeXMLNS', active: true },
  ],
}

const svg = `
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 0 0">
  <defs>
    <path d=""></path>
    <path d=""></path>
  </defs>
</svg>
`

const result = optimize(svg, options);

console.dir(result);
```

1. Install svgo 2.7.0
2. Run the script
3. Notice a result is logged
4. Install svgo 2.8.0
5. Run the script
6. Notice an error is logged

The following will make the error go away:

- Set `multipass: false`
- Disable the `reusePaths` plugin
- Disable the `removeXMLNS` plugin

**Expected behavior**

Used namespaces aren’t removed by `removeXMLNS`, so the plugins can work together.

**Screenshots**

N/A

**Desktop (please complete the following information):**
 - SVGO Version: 2.8.0
 - NodeJs Version 16.13.0
 - OS: Pop!_OS 21.04
 
**Additional context**

The error was introduced in version 2.8.0.

**Comment by TrySound** [comment]
Yeah, parser indeed doesn't understand unbound namespace. This was unexpecteed. But anyway, at the moment only convertTransform may produce different result with multipass. And we gonna make it more deterministic. So I recommend to disable multipass mode. It will be removed in v3.

**Comment by cyberalien** [comment]
> It will be removed in v3.

Still bugged in 3.0.2

**Comment by SethFalco** [comment]
This has been resolved and will be released in SVGO v3.0.4.

* `removeXMLNS` will no longer remove the XLink namespace at all.
* A new plugin, `removeXlink` can be used to migrate from the XLink namespace to the SVG 2 equivalent when only SVG 2 or inline HTML compatibility is required.
